src/rwlock.c

```diff
@@ -147,15 +147,13 @@ int pthread_rwlock_tryrdlock(pthread_rwlock_t *_lock)
   lock = *_lock;
 
   pthread_mutex_lock(&lock->mutex);
-  if (lock->num_active < 0)
+  if (lock->num_active == 0)

```

Comment:
It is (and was) allowed to have multiple readers, it should be >= 0.
```suggestion
  if (lock->num_active >= 0)
```